src/main/resources/default/templates/biz/process/process.html.pasta

```diff
@@ -148,7 +149,9 @@
         <script type="text/javascript">
             sirius.ready(function () {
                 setTimeout(function () {
-                    window.location.reload();
+                    if (___isAutoRefreshEnabled) {
+                        window.location.reload();
+                    }
```

Comment:
This solution is static, meaning if 'isAutoRefreshEnabled' is false, it will stay false (without a page reload) and the time out block spins around doing nothing.

I think we could have an even better user experience and simpler code by simply storing the blockage of th refresh in a simple JS boolean field and control the behaviour of the refresh with that.

Also, if we write the setting into the URL parameters (e.g. ".../...?refresh=false") we could propagate the setting when filtering the messages or switching the page in order to not reload even though we are still practically from a user point of view (even if there are page reloads) on the same page.

Reply:
> This solution is static, meaning if 'isAutoRefreshEnabled' is false, it will stay false (without a page reload) and the time out block spins around doing nothing.
- In that case the `setTimout` block will be triggered exactly **once** without doing anything, but I agree. In that case it might be better to move the `if` statement a few lines up, in order to prevent it triggering even once.
- Note: The complete <script> tag is only added if `<i:if test="process.shouldAutorefresh()">`
- Note: Take a look at how the reload worked before. I basically only added one `if` statement.

> I think we could have an even better user experience [...]
- How does your suggestion affect the user experience? Currently, after pressing "disable auto refresh" the page stops auto refreshing. Therefore it does exactly what a user would expect.

> [...] and simpler code by simply storing the blockage of th refresh in a simple JS boolean field and control the behaviour of the refresh with that.
- I thougt about solving it with pure JS at first too, but @sabieber did not like the solution, or at least suggested that I solve it as I did here.

> Also, if we write the setting into the URL parameters (e.g. ".../...?refresh=false") we could propagate the setting when filtering the messages or switching the page in order to not reload even though we are still practically from a user point of view (even if there are page reloads) on the same page.
- Could you please explain this further, I did not get your point. (Gern auch in deutsch). :)

Reply:
@fhaScireum I suggested the parameter solution as it would stick around even when the user actively reloads the page (which he will after deactivating the auto refresh). If there are arguments against that we can change it of course.